api/api.go

```diff
@@ -145,6 +145,9 @@ func NewAPIWithVersion(ctx context.Context, globalConfig *conf.GlobalConfigurati
 			r.Use(api.requireAuthentication)
 			r.Get("/", api.UserGet)
 			r.With(sharedLimiter).Put("/", api.UserUpdate)
+			r.Route("/email", func(r *router) {
```

Comment:
Thanks for review, and comments, actually at the beginning I just added a new parameter to /user endpoint, then saw your comment and changed again https://github.com/supabase/gotrue/issues/361#issuecomment-1024919581, I am using same function `api.UserUpdate`, because most of part would be same, I didn't wanna duplicate many lines.